### PR TITLE
Migrate to laravel 5.2 version

### PR DESCRIPTION
**Pull request by avengerweb**
Migrate to laravel 5.2 version
Adding some needed env parameters:
APP_ENV=production // this need for some service providers
APP_DEBUG=false // easy going to debug, needed on install
APP_KEY=SomeRandomString // **Need for encrypt session and etc.** that should be unique!
**Update 2:**
Fixed home page
Move CORS headers to right place in Laravel Middlewares

### CODE REVIEW

**Comment by Helias** [comment]
Thanks for pull request, we will merge it as soon as possible

**Comment by avengerweb** [comment]
Ok

**Comment by avengerweb** [comment]
So, in last commit I was added something in laravel style. Now, I will wait merge, or maybe discussions.

File: public/index.php [hunk 1]
```diff
@@ -46,15 +47,12 @@
...
-$kernel = $app->make('Illuminate\Contracts\Http\Kernel');
+$kernel = $app->make(Illuminate\Contracts\Http\Kernel::class);
 
 $response = $kernel->handle(
 	$request = Illuminate\Http\Request::capture()
 );
 
-// CORS fix
-header('Access-Control-Allow-Origin: *');
-
```
Comment (FrancescoBorzi) [inline public/index.php:57]: why did you remove this?

File: .env.example [hunk 2]
```diff
@@ -1,7 +1,11 @@
...
+APP_KEY=SomeRandomString
+
 DB_HOST=localhost
 DB_WORLD=world
 DB_CHARACTERS=characters
 DB_AUTH=auth
 DB_HOTFIXES=hotfixes
 DB_USERNAME=root
-DB_PASSWORD=password
+DB_PASSWORD=password
```
Comment (FrancescoBorzi) [inline .env.example]: no new line at the end of file

File: .gitignore [hunk 3]
```diff
@@ -1,4 +1,9 @@
+.idea
 /vendor
 /node_modules
 .env
 .DS_Store
+composer.lock
+/public/storage
+Homestead.yaml
+Homestead.json
```
Comment (FrancescoBorzi) [inline .gitignore]: no newline at the end of file

File: config/database.php [hunk 4]
```diff
@@ -44,76 +44,76 @@
...
+            'driver'        => 'mysql',
+            'host'            => env('DB_HOST', 'localhost'),
+            'database'    => env('DB_WORLD', 'world'),
+            'username'    => env('DB_USERNAME', 'root'),
+            'password'    => env('DB_PASSWORD', 'password'),
+            'charset'     => 'utf8',
+            'collation' => 'utf8_unicode_ci',
+            'prefix'        => '',
+            'strict'        => false,
+        ],
```
Comment (FrancescoBorzi) [inline config/database.php:59]: why this indentation?

File: public/index.php (hunk 1 above)
Comment (avengerweb) [inline public/index.php:57]: Moved here https://github.com/avengerweb/TC-JSON-API/blob/master/app/Http/Middleware/CORSAccess.php#L29

File: config/database.php (hunk 4 above)
Comment (avengerweb) [inline config/database.php:59]: 4 spaces, not 2

File: public/index.php (hunk 1 above)
Comment (FrancescoBorzi) [inline public/index.php:57]: ok, all those indentation changes are needed?

**Comment by FrancescoBorzi** [comment]
@avengerweb thanks a lot of this PR, can you check the merge conflicts?

**Comment by avengerweb** [comment]
Resolved

File: public/index.php (hunk 1 above)
Comment (avengerweb) [inline public/index.php:57]: Yes, http://www.php-fig.org/psr/psr-2/

**Comment by avengerweb** [comment]
Any news?

**Comment by Helias** [comment]
I downloaded your TC-JSON-API and "composer install" gives me this error:

http://s7.postimg.org/spkg3h123/sdds.png

**Comment by avengerweb** [comment]
Try to composer update or remove composer.lock

**Comment by Helias** [comment]
Ok, we'll merge your PR soon ;)

I saw that you moved some achievements route in a controller, I think that we should do this with all routes.

File: app/Http/Controllers/Achievements/AchievementsController.php [hunk 5]
```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: AvengerWeb
+ * Date: 25.03.16
+ * Time: 22:37
+ */
```
Comment (Helias) [inline app/Http/Controllers/Achievements/AchievementsController.php]: Can you remove this automatically generated comments?

Comment (avengerweb) [inline app/Http/Controllers/Achievements/AchievementsController.php]: Done

**Comment by avengerweb** [comment]
Removed

**Comment by Helias** [comment]
Merged! Thanks for the PR! ;)
